Write more comments.

sunfishcode opened the pull request:
Make a pass over the crate adding and cleanup up comments and doc aliases.

cgwalters reviewed (approved):
With this can we `#![deny(missing_docs)]`?

sunfishcode commented:
Thanks for bringing that up! We already had a `deny(missing_docs)`, though we previously had a few `allow(missing_docs)` local overrides, but this PR got pretty close to removing the last of the overrides, so I've now gone ahead and added the last few comments and removed the last overrides.
